app/services/concerns/xccdf_report/rules.rb

```diff
@@ -73,15 +84,40 @@ def new_rules
         end
       end
 
+      def save_rule_references
+        return @rule_references if @rule_references
+
+        @rule_references = []
+        new_rules.map do |rule|
+          @rule_references << RuleReference.from_oscap_objects(rule.references)
+        end
+        RuleReference.import(@rule_references.flatten,
+                             columns: %i[href label],
+                             ignore: true)
+      end
+
       def save_rules
         new_profiles = Profile.where(ref_id: profiles.keys)
         add_profiles_to_old_rules(rules_already_saved, new_profiles)
-        Rule.import!(
-          new_rules.each_with_object([]) do |rule, new_rules|
-            new_rule = Rule.new(profiles: new_profiles).from_oscap_object(rule)
-            new_rules << new_rule
-          end, recursive: true
-        )
+        new_rule_records = new_rules
+                           .each_with_object([]).map do |oscap_rule, _new_rules|
+          rule_object = Rule.new(profiles: new_profiles).from_oscap_object(oscap_rule)
+          if oscap_rule.identifier
+            rule_object.rule_identifier = RuleIdentifier
+                                          .new(label: oscap_rule.identifier, system: oscap_rule.identifier_system)
+          end
+          rule_object
```

Comment:
How about creating `RuleReferencesRule` in here instead of through ActiveRecord in `associate_rule_references`?

Reply:
I tried, but active-record-import was complaining (I forget the error message). I can try again, but there is nothing in active-record-import that claims join tables are supported for `has_many through: :foo`.